uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/installer/PlayStore.java

```diff
@@ -187,4 +187,49 @@ public void installApp(@NonNull final String searchHint) {
         installAppFromMarketPage();
     }
 
+    public void updateApp(@NonNull final String searchHint) {
+        if (isStringPackageName(searchHint)) {
+            launchMarketPageForPackage(searchHint);
+        } else {
+            searchAppOnGooglePlay(searchHint);
+            selectGooglePlayAppFromAppName();
+        }
+
+        updateAppFromMarketPage();
+    }
+
+    private void updateAppFromMarketPage() {
+        Logger.i(TAG, "Update App From Market Page..");
+        try {
+            updateAppFromMarketPageInternal();
+        } catch (final UiObjectNotFoundException e) {
+            acceptGooglePlayTermsOfService();
+            try {
+                updateAppFromMarketPageInternal();
+            } catch (UiObjectNotFoundException ex) {
+                throw new AssertionError(e);
+            }
+        }
+    }
```

Comment:
Looks like in this entire block, the only thing different from `install` is `line 204` i.e. `updateAppFromMarketPageInternal`. Everything else is same as `install` right? I think there is definitely opportunity for code sharing.

Reply:
Line 204 and 208. Rest is just calling updateAppFromMarketPageInternal and catching exception.

Reply:
Yes, so I think definitely opportunity for code sharing here